### PR TITLE
[SYCL-MLIR][LoopInternalization] Only optimize innermost loop

### PR DESCRIPTION
**Pull request by whitneywhtsang**
Tile changes the access order in a multi level loop nest. 
As the motivating test case is a single level loop nest, only optimize innermost loop in a perfect loop nest for the time being.

### CODE REVIEW

**Review by sommerlukas — APPROVED** [review]

**Comment by etiotto** [comment]
This PR limits the transformation to a single loop. How about we tile only the innermost loop in a perfect loop nest instead.

**Comment by whitneywhtsang** [comment]
> This PR limits the transformation to a single loop. How about we tile only the innermost loop in a perfect loop nest instead.

That should work too, and less restrictive, changed.

File: polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp [hunk 1]
```diff
@@ -91,37 +91,24 @@ bool isCandidate(LoopLikeOpInterface loop) {
...
-        builder.create<arith::ConstantIndexOp>(builder.getUnknownLoc(), 1);
-    tileSizes.resize(nestedLoops.size(), one);
-  }
+  return builder.create<arith::ConstantIndexOp>(builder.getUnknownLoc(),
+                                                LoopInternalizationTileSize);
 }
 
-LogicalResult tile(MutableArrayRef<affine::AffineForOp> nestedLoops,
-                   ArrayRef<Value> tileSizes,
+LogicalResult tile(affine::AffineForOp nestedLoops, Value tileSizes,
```
Comment (etiotto) [inline polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp:100]: tileSizes -> tileSize

Comment (etiotto) [reply polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp:100]: nestedLoops -> innerLoop (or simply loop)

File: polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp [hunk 2]
```diff
@@ -91,37 +91,24 @@ bool isCandidate(LoopLikeOpInterface loop) {
...
+      tilePerfectlyNestedParametric({nestedLoops}, tileSizes, &newNestedLoops);
+  tiledNest = SmallVector<affine::AffineForOp>(newNestedLoops.begin() + 1,
+                                               newNestedLoops.end());
   return res;
 }
-LogicalResult tile(ArrayRef<scf::ForOp> nestedLoops, ArrayRef<Value> tileSizes,
+LogicalResult tile(scf::ForOp nestedLoops, Value tileSizes,
                    SmallVectorImpl<scf::ForOp> &tiledNest) {
-  tiledNest = tile(nestedLoops, tileSizes, nestedLoops.back());
+  tiledNest = tile({nestedLoops}, tileSizes, nestedLoops);
```
Comment (etiotto) [inline polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp:111]: Similar comments as line 100

**Review by etiotto — APPROVED** [review]
LGTM after comments are addressed.

File: polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp]: done.

**Review by etiotto — APPROVED** [review]

**Pull request merged**